### ISSUE TITLE
[PPC] Unit test failure in SimG4Core/DD4hepGeometry

### ISSUE THREAD

**Issue by mrodozov** [cms-sw/cmssw#32090]
This unit test is failing:
DD4hepGeometryTestDriver
in 
SimG4Core/DD4hepGeometry
running
`cmsRun /scratch/cmsbuild/jenkins_a/workspace/ib-run-qa/CMSSW_11_2_X_2020-11-05-2300/src/SimG4Core/DD4hepGeometry/test/python/testG4Geometry.py`

```
Current Modules:
Module: DD4hepTestG4Geometry:test (crashed)
```

https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_ppc64le_gcc820/CMSSW_11_2_X_2020-11-05-2300/unitTestLogs/SimG4Core/DD4hepGeometry#/93-93

**Comment by makortel** [comment]
assign simulation, geometry

**Comment by makortel** [comment]
More details from the log in the description
```
#3  0x0000100009924838 in sig_dostack_then_abort () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginFWCoreServicesPlugins.so
#4  <signal handler called>
#5  0x00001000029e73d8 in __kernel_sincosl () from /lib64/libm.so.6
#6  0x00001000029dffa8 in sincosl@@GLIBC_2.17 () from /lib64/libm.so.6
#7  0x000010022e4d4f8c in algorithm(dd4hep::Detector&, cms::DDParsingContext&, dd4hep::xml::Handle_t) [clone .isra.136] () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDD4hep_GeometryEcalCommonDataPlugins.so
#8  0x0000100228bffad8 in void dd4hep::xml::Collection_t::for_each<dd4hep::Converter<dd4hep::(anonymous namespace)::DDLAlgorithm, dd4hep::xml::Handle_t> >(dd4hep::Converter<dd4hep::(anonymous namespace)::DDLAlgorithm, dd4hep::xml::Handle_t>) const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDD4HepPlugins.so
#9  0x0000100228c0d6b8 in load_dddefinition(dd4hep::Detector&, dd4hep::xml::Handle_t) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDD4HepPlugins.so
#10 0x0000100228c0ec34 in (anonymous namespace)::Factory<dd4hep::(anonymous namespace)::xml_document_DDDefinition, long (dd4hep::Detector*, dd4hep::xml::Handle_t*)>::call(dd4hep::Detector*, dd4hep::xml::Handle_t*) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDD4HepPlugins.so
#11 0x000010001f546668 in dd4hep::DetectorLoad::processXMLElement(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dd4hep::xml::Handle_t const&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#12 0x000010001f5473ac in dd4hep::DetectorLoad::processXML(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dd4hep::xml::UriReader*) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#13 0x000010001f53b3bc in non-virtual thunk to dd4hep::DetectorImp::fromCompact(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dd4hep::DetectorBuildType) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#14 0x0000100228b79444 in (anonymous namespace)::Factory<dd4hep::(anonymous namespace)::DD4hep_CompactLoader, long (dd4hep::Detector*, int, char**)>::call(dd4hep::Detector*, int, char**) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCorePlugins.so.1.14
#15 0x000010001f53bb48 in long dd4hep::PluginService::Create<long, dd4hep::Detector*, int, char**>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dd4hep::Detector*, int, char**) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#16 0x000010001f5340e8 in dd4hep::DetectorImp::apply(char const*, int, char**) const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/external/slc7_ppc64le_gcc820/lib/libDDCore.so.1.14
#17 0x000010001ee98144 in cms::DDDetector::process(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDDCMS.so
#18 0x000010001ee98b44 in cms::DDDetector::DDDetector(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libDetectorDescriptionDDCMS.so
#19 0x0000100221c56420 in DDDetectorESProducer::produceGeom(IdealGeometryRecord const&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#20 0x0000100221c64d84 in decltype ({parm#1}()) edm::convertException::wrap<edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}::operator()() const::{lambda()#1}>(edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}::operator()() const::{lambda()#1}) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#21 0x0000100221c652a8 in edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}::operator()() const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#22 0x0000100221c66684 in void edm::SerialTaskQueueChain::actionToRun<edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}&>(edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#23 0x0000100221c66788 in edm::SerialTaskQueue::QueuedTask<edm::SerialTaskQueueChain::push<edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}>(edm::eventsetup::Callback<DDDetectorESProducer, std::unique_ptr<cms::DDDetector, std::default_delete<cms::DDDetector> >, IdealGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<IdealGeometryRecord> >::runProducerAsync(std::__exception_ptr::exception_ptr const*, edm::eventsetup::EventSetupRecordImpl const*, edm::EventSetupImpl const*, edm::ServiceToken const&)::{lambda()#1}&&)::{lambda()#1}>::execute() () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDetectorDescriptionPlugins.so
#24 0x00001000024ab780 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::process_bypass_loop (this=0x1000037ff200, context_guard=..., t=0x1000037f2440, isolation=0) at ../../include/tbb/task.h:1003
#25 0x00001000024abaa4 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::local_wait_for_all (this=0x1000037ff200, parent=..., child=<optimized out>) at ../../include/tbb/task.h:1003
#26 0x00001000024ac4b0 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::wait_for_all (this=<optimized out>, parent=..., child=0x0) at ../../src/tbb/governor.h:123
#27 0x00001000001de3c0 in tbb::interface7::internal::delegated_function<edm::eventsetup::DataProxy::get(edm::eventsetup::EventSetupRecordImpl const&, edm::eventsetup::DataKey const&, bool, edm::ActivityRegistry const*, edm::EventSetupImpl const*) const::{lambda()#1} const, void>::operator()() const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#28 0x00001000024a4be8 in tbb::interface7::internal::task_arena_base::internal_execute (this=0x1000004e6ce0 <edm::esTaskArena()::s_arena>, d=...) at ../../src/tbb/arena.cpp:1105
#29 0x00001000001de65c in edm::eventsetup::DataProxy::get(edm::eventsetup::EventSetupRecordImpl const&, edm::eventsetup::DataKey const&, bool, edm::ActivityRegistry const*, edm::EventSetupImpl const*) const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#30 0x000010000028a468 in edm::eventsetup::EventSetupRecordImpl::getFromProxy(edm::eventsetup::DataKey const&, edm::eventsetup::ComponentDescription const*&, bool, edm::EventSetupImpl const*) const () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#31 0x00001002227bb554 in DD4hepTestG4Geometry::analyze(edm::Event const&, edm::EventSetup const&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/pluginDD4hepGeometryTestPlugins.so
#32 0x00001000003e9948 in edm::one::EDAnalyzerBase::doEvent(edm::EventTransitionInfo const&, edm::ActivityRegistry*, edm::ModuleCallingContext const*) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#33 0x00001000003c0d24 in edm::WorkerT<edm::one::EDAnalyzerBase>::implDo(edm::EventTransitionInfo const&, edm::ModuleCallingContext const*) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#34 0x00001000002dc7cc in decltype ({parm#1}()) edm::convertException::wrap<edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}>(edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#35 0x00001000002dca68 in bool edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#36 0x00001000002dced8 in std::__exception_ptr::exception_ptr edm::Worker::runModuleAfterAsyncPrefetch<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(std::__exception_ptr::exception_ptr const*, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#37 0x00001000002e0494 in void edm::SerialTaskQueueChain::actionToRun<edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute()::{lambda()#1}&>(edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute()::{lambda()#1}&) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#38 0x00001000002e06b8 in edm::SerialTaskQueue::QueuedTask<edm::SerialTaskQueueChain::push<edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute()::{lambda()#1}&>(edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute()::{lambda()#1}&)::{lambda()#1}>::execute() () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-05-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
```

**Comment by mrodozov** [comment]
this is fixed now by #33705
